src/DynamoCore/Linting/LinterManager.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Core;
+using Dynamo.Extensions;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Linting.Interfaces;
+using Dynamo.Linting.Rules;
+using Dynamo.Models;
+
+namespace Dynamo.Linting
+{
+    public class LinterManager : NotificationObject
+    {
```

Comment:
Also I see it is possible to remove an extension during the session ( only when an extension gets uninstalled or upgraded ? ). We should handle this case and remove the LinterExtension descriptor and call dispose on it as needed.

Reply:
Disregard the above, we don't have to worry about this for now